R/link_ch_with_adms.R

```diff
@@ -0,0 +1,161 @@
+    # ..... this is the date we want to test against the two care home dates
+    dplyr::mutate(test_date = dplyr::coalesce(cij_start_date, record_keydate2))
+  # populate with care home data ...
+  # ... if the test_date falls within the care home window
+
+  # link ch to ae
+  link_ae <- adms %>%
+    dplyr::filter(recid == "AE2") %>%
+    dplyr::inner_join(
+      care_home_data,
+      by = dplyr::join_by(
+        x$anon_chi == y$anon_chi,
+        x$test_date > y$record_keydate1,
+        x$test_date < y$record_keydate2
+      ),
+      na_matches = "never",
+      unmatched = "drop",
+      multiple = "first",
+      suffix = c("", ".y")
+    ) %>%
+    dplyr::mutate(
+      ch_name = dplyr::coalesce(ch_name, ch_name.y),
+      ch_postcode = dplyr::coalesce(ch_postcode, ch_postcode.y)
+    ) %>%
+    dplyr::select(-c(
+      dplyr::ends_with(".y"),
+      "test_date"
+    )) %>%
+    dplyr::mutate(
+      sc_ch_link_ae = 1L,
+      sc_ch_link_adms = NA
+    )
+
+  # link ch to 01b, 04b, gls
+  link_adms <- adms %>%
+    dplyr::filter(recid %in% c("01B", "04B", "GLS")) %>%
+    dplyr::inner_join(
+      care_home_data,
+      by = dplyr::join_by(
+        x$anon_chi == y$anon_chi,
+        x$test_date > y$record_keydate1,
+        x$test_date < y$record_keydate2
+      ),
+      na_matches = "never",
+      unmatched = "drop",
+      multiple = "first",
+      suffix = c("", ".y")
+    ) %>%
+    dplyr::mutate(
+      ch_name = dplyr::coalesce(ch_name, ch_name.y),
+      ch_postcode = dplyr::coalesce(ch_postcode, ch_postcode.y)
+    ) %>%
+    dplyr::select(-c(
+      dplyr::ends_with(".y"),
+      "test_date"
+    )) %>%
+    dplyr::mutate(
+      sc_ch_link_adms = 1L,
+      sc_ch_link_ae = NA
+    )
+
+
+  # Now update ch_name and ch_postcode values in ep from link_ae
+  ep <- ep %>%
+    dplyr::left_join(
+      rbind(link_ae, link_adms) %>%
+        dplyr::select(
+          ep_row_id_CE,
+          ch_name,
+          ch_postcode,
+          sc_ch_link_ae,
+          sc_ch_link_adms
+        ),
+      by = "ep_row_id_CE",
+      suffix = c("", ".y")
+    ) %>%
+    dplyr::mutate(
+      ch_name = dplyr::coalesce(ch_name, ch_name.y),
+      ch_postcode = dplyr::coalesce(ch_postcode, ch_postcode.y)
+    ) %>%
+    dplyr::select(-"ep_row_id_CE", -tidyselect::ends_with(".y")) %>%
+    # Standardise sc_ch_link_ae, sc_ch_link_adms.
+    # `sc_ch_link_ae` should be:
+    #   1L if someone goes to ae from ch, (recid == AE)
+    #   0L if someone goes to ae but not from ch, (recid == AE)
+    #   NA if recid != AE, meaning this variable does not apply
+    dplyr::mutate(
+      sc_ch_link_ae = dplyr::if_else((is.na(sc_ch_link_ae) &
+        recid == "AE2"),
+      0L, sc_ch_link_ae
+      ),
+      sc_ch_link_adms = dplyr::if_else((
+        is.na(sc_ch_link_adms) &
+          recid %in% c("01B", "04B", "GLS")
+      ),
+      0L, sc_ch_link_adms
+      )
+    )
```

Comment:
Just wondering about this last part, should the final flags show 1, 0,  NA? or just 1, 0?

The code seems to be working apart from the one error i get above with hscp. But i am wondering about the final output. In the episode file, 01b records have NA with `sc_ch_link_ae` is that right?

Reply:
The final flags should show 1, 0, NA. It is the best practice. Imagine a general scenario -- if we assign zero to those which are supposed to NA and we need to get an average of the variable, the descriptive statistics will be misleading.

sc_ch_link_ae is supposed to be applied to AE2 only while sc_ch_link_adms is supposed to applied to 01B, 04B, and GLS. So, 01B records should have NA with sc_ch_link_ae while AE2 records should have NA with sc_ch_link_adms.